packages/fuel-indexer-graphql-dyn/src/store/store_type_builder.rs

```diff
@@ -0,0 +1,324 @@
+use super::assoc_type::*;
+use super::data_type::*;
+use super::meta::*;
+use super::obj_type::*;
+use super::self_prelude::*;
+use super::store_type::*;
+
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+pub struct StoreTypeBuilder {
+    pub data: IndexMap<DataTypeId, DataTypeBuilder>,
+    pub obj: IndexMap<ObjTypeId, ObjTypeBuilder>,
+    pub assoc: IndexMap<AssocTypeId, AssocTypeBuilder>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum DataTypeBuilder {
+    Composite(StructTypeBuilder),
+    Enum(EnumTypeBuilder),
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct StructTypeBuilder {
+    pub name: Name,
+    pub fields: IndexMap<DataFieldId, DataFieldType>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct EnumTypeBuilder {
+    pub name: Name,
+    pub variants: IndexMap<DataFieldId, DataFieldType>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ObjTypeBuilder {
+    pub name: Name,
+    pub fields: IndexMap<DataFieldId, DataFieldType>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct AssocTypeBuilder {
+    pub name: Name,
+    pub tail: ObjTypeId,
+    pub head: ObjTypeId,
+    pub fields: IndexMap<DataFieldId, DataFieldType>,
+}
+
+impl StoreTypeBuilder {
+    pub fn data(
+        &mut self,
+        name: impl Into<String>,
+    ) -> (DataTypeId, &mut DataTypeBuilder) {
+        let name = Name::new_pascal(name);
+        self.data
+            .iter_mut()
+            .find(|(_, data)| data.name() == name)
+            .map(|(id, data)| (id.clone(), data))
+            .unwrap_or_else(|| panic!("Data not found: {}", name))
```

Comment:
Anyway we can fail gracefully here with `Result` rather than `panic` ? (similar thoughts for the rest of the `panic` in this module)